GSoC'21: New Hindi Branch

SanjaySinghRajpoot opened the pull request:
I have created this new Hindi branch, as the other hindi branch was outdated. Hindi branch was not updated for a long time, many changes were missing from the hindi branch including new text properties and code runners. Many web pages were missing too. 

This new main-hindi branch is forked from the latest main branch of p5.js-website. The main was cloned first and then hindi language was added as per the documentation. Then a new branch called main-Hindi was created to compose all the changes in one place.

It has all new changes for all the languages. I have added Hindi translation to this branch so that it can be easily merged into the main branch. I have accumulated all the new changes to this PR so that it is easier to track any new error or to make any new changes to the main-hindi branch.

This main-hindi branch is made by merging the PR's which were verified by the mentor and all the new features were checked and tested on a local machine.  

Major Changes: 
- Adding a new Teach page for p5-website.
- Adding CSS to Copy button for Get Started Page.
- Get Started Page Updated to the latest format.
- New books were added to Books Page with proper styling and images.
- Translated and Checked Community Page, Books Page, Get Started Page, and Teach Page.
- All the Examples are now created for the Hindi language.
- All References were translated and modified with new text props.

PR's
- https://github.com/processing/p5.js-website/pull/1060
- https://github.com/processing/p5.js-website/pull/1041
- https://github.com/SanjaySinghRajpoot/p5.js-website/pull/18
- https://github.com/SanjaySinghRajpoot/p5.js-website/pull/17

New Website in Action 
video - https://www.youtube.com/watch?v=dcRnZSXqKGQ

An outcome of Google Summer of Code 2021
Author- Sanjay Singh Rajpoot @SanjaySinghRajpoot
Mentor - Aditya Rana @ranaaditya

ranaaditya commented:
Hi @outofambit @Qianqianye , this Pull Request directly aims to add **Hindi** language (Internationalization of p5 website) to the [p5.js-webiste](https://github.com/processing/p5.js-website).

@SanjaySinghRajpoot  and I have contributed significant amount of hard work and time to it. And this PR directly aims to the @SanjaySinghRajpoot 's **GSoC'21 Project**, So I request the current active maintainers to please give it a review and try to merge it iff possible now.

@SanjaySinghRajpoot has also made a live demo of this PR / work his [YouTube Channel](https://youtu.be/dcRnZSXqKGQ) to demonstrate his work.

In the meantime @SanjaySinghRajpoot  is also trying to host this website with Hindi translation (his GSoC work) on his personal domain / website so that we can have a look of what he has built so far and this may help maintainers to review his work also !

Thanks,
Aditya Rana
GSoC Mentor

outofambit commented:
hey @ranaaditya @SanjaySinghRajpoot the changes in this PR look good to me, but i'm confused about #1060. It's still open and seems to be targetting this branch. Do we need to do anything with it? It looks like it might have some changes that aren't reflected here. Let me know please and then I can proceed with merging!

ranaaditya commented:
> hey @ranaaditya @SanjaySinghRajpoot the changes in this PR look good to me, but i'm confused about #1060. It's still open and seems to be targetting this branch. Do we need to do anything with it? It looks like it might have some changes that aren't reflected here. Let me know please and then I can proceed with merging!

Hi @outofambit , So we were working previously on [hindi branch](https://github.com/processing/p5.js-website/tree/hindi) but it was so outdated and didn't have latest changes so we decided to branch out new [main-hindi branch](https://github.com/sanjaySinghRajpoot/p5.js-website/tree/main-hindi) from latest master and add all the previous done work / commits in this new branch.

This PR aims to add **Hindi Translation** to main branch of this repo and other PR (open one) that aims to add hindi translation to old [hindi branch](https://github.com/processing/p5.js-website/tree/hindi). This PR has everything which is needed for Hindi translation.

Thanks !

limzykenneth commented:
Does the old `hindi` branch still needs to be kept up to date now that we have this `main-hindi` branch? If there's no other need for the old `hindi` branch can we remove it?

I see that the `hi` link has been added to the header already which means that when this is merged into `main` the link to the Hindi version of the website will be live. @outofambit You may want to time this with some announcement? Just noting.

For the website to build fully `package.json` also needs to be updated [here](https://github.com/processing/p5.js-website/blob/main/package.json#L5) by adding the language code for `Hindi`.

I'll handle the future porting of the Hindi reference translation over to Pontoon now that I've merge #1069 and there shouldn't be a merge conflict unlike #1062 (which builds on existing files) so don't worry about it.

ranaaditya commented:
> Does the old `hindi` branch still needs to be kept up to date now that we have this `main-hindi` branch? If there's no other need for the old `hindi` branch can we remove it?
> 
> I see that the `hi` link has been added to the header already which means that when this is merged into `main` the link to the Hindi version of the website will be live. @outofambit You may want to time this with some announcement? Just noting.
> 
> For the website to build fully `package.json` also needs to be updated [here](https://github.com/processing/p5.js-website/blob/main/package.json#L5) by adding the language code for Hindi.
> 
> I'll handle the future porting of the Hindi reference translation over to Pontoon now that I've merge #1069 and there shouldn't be a merge conflict unlike #1062 (which builds on existing files) so don't worry about it.

Hi @limzykenneth  we don't need old Hindi branch now, since we have new main-hindi branch which has all changes of old Hindi branch.

SanjaySinghRajpoot commented:
> hey @ranaaditya @SanjaySinghRajpoot the changes in this PR look good to me, but i'm confused about #1060. It's still open and seems to be targetting this branch. Do we need to do anything with it? It looks like it might have some changes that aren't reflected here. Let me know please and then I can proceed with merging!

we don't need that PR anymore as I have already merged it into the new main-hindi branch. All the changes are in this single PR for this new branch. The PR #1060 is made for the old hindi branch.

SanjaySinghRajpoot commented:
> Does the old `hindi` branch still needs to be kept up to date now that we have this `main-hindi` branch? If there's no other need for the old `hindi` branch can we remove it?
> 
> I see that the `hi` link has been added to the header already which means that when this is merged into `main` the link to the Hindi version of the website will be live. @outofambit You may want to time this with some announcement? Just noting.
> 
> For the website to build fully `package.json` also needs to be updated [here](https://github.com/processing/p5.js-website/blob/main/package.json#L5) by adding the language code for Hindi.
> 
> I'll handle the future porting of the Hindi reference translation over to Pontoon now that I've merge #1069 and there shouldn't be a merge conflict unlike #1062 (which builds on existing files) so don't worry about it.

We may remove the `old hindi` branch once this branch is merged. I think once the code is in production we will not require `old hindi` branch at all. 

I have updated the `package.json` file for the Hindi language and also updated the Repo.

outofambit commented:
thank you @SanjaySinghRajpoot, @ranaaditya, and @limzykenneth! i'm going to merge this (and close #1060).

i'll make sure we talk about this on our social media accounts soon, we might wait for the GSoC writeup so we can reference that in those posts. Thank you again, this is very exciting!! 🎉🎉🎉

ranaaditya commented:
> thank you @SanjaySinghRajpoot, @ranaaditya, and @limzykenneth! i'm going to merge this (and close #1060).
> 
> i'll make sure we talk about this on our social media accounts soon, we might wait for the GSoC writeup so we can reference that in those posts. Thank you again, this is very exciting!! tadatadatada

Thanks a lot @outofambit for merging the work :partying_face: . 

I can see now [Hindi option in p5js.org](https://p5js.org/hi) website.

Again good work @SanjaySinghRajpoot :clap: keep it up !

@outofambit  I will take care if any translation or anything related to Hindi branch will come in future.

Thanks !

SanjaySinghRajpoot commented:
Wow !!. It looks amazing all my GSoC work is now live 🥳🥳. 

Thank You @ranaaditya  @outofambit @limzykenneth and the entire Processing team for this amazing project. You guys are remarkable.

If there are any issues regarding this new branch, I would love to fix them around. 

Thank You ❤️

rakesh2OO5 commented:
Hi folks! 👋 Just a gentle ping on this — all checks are passing and I’d love any feedback when you have time.

@ksen0 since you raised the original issue, would really appreciate your thoughts if you get a chance.

Thanks for maintaining the project!

rakesh2OO5 commented:
Hi folks 👋 — just following up as this PR has been open for about a month now.

This adds optional tab indentation with accessible keyboard handling, which improves usability for people using the website editor with assistive tech. All checks are passing, and I’m happy to rebase or adjust anything if needed.

If this isn’t a priority right now, no worries — just let me know. I’d really appreciate any quick feedback when you have a moment.

Thanks again for maintaining p5.js! 🙏

Nwakaego-Ego commented:
Hi @rakesh2OO5, just flagging that to work on an issue you first need to be assigned by a maintainer or steward 
before opening a PR. Opening a PR without being assigned goes against the [community guidelines](https://github.com/processing/p5.js-website/blob/main/README.md). Thank you.

CC @doradocodes @ksen0
